src/python/gudhi/nerve_gic.pyx

```diff
@@ -400,23 +408,726 @@ cdef class CoverComplex:
+                    list_preimage.append(list(range(start, end+1)))
+                list_preimage = list(itertools.product(*list_preimage))
+                for pre_idx in list_preimage:
+                    try:
+                        binned_data[pre_idx].append(i)
+                    except KeyError:
+                        binned_data[pre_idx] = [i]
+
+        # Initialize the cover map, that takes a point and outputs the clusters to which it belongs
+        cover, clus_base = [[] for _ in range(num_pts)], 0
+
+        # For each patch
+        for preimage in binned_data:
+
+            # Apply clustering on the corresponding subpopulation
+            idxs = np.array(binned_data[preimage])
+            if len(idxs) > 1:
+                clusters = self.clustering.fit_predict(X[idxs,:]) if self.input == "point cloud" else self.clustering.fit_predict(X[idxs,:][:,idxs])
+            elif len(idxs) == 1:
+                clusters = np.array([0])
+            else:
+                continue
+
+            # Collect various information on each cluster
+            num_clus_pre = np.max(clusters) + 1
+            for clus_i in range(num_clus_pre):
+                node_name = clus_base + clus_i
+                subpopulation = idxs[clusters == clus_i]
+                self.node_info_[node_name] = {}
+                self.node_info_[node_name]["indices"] = subpopulation
+                self.node_info_[node_name]["size"] = len(subpopulation)
+                self.node_info_[node_name]["colors"] = np.mean(self.colors[subpopulation,:], axis=0)
+                self.node_info_[node_name]["patch"] = preimage
+
+            # Update the cover map
+            for pt in range(clusters.shape[0]):
+                node_name = clus_base + clusters[pt]
+                if clusters[pt] != -1 and self.node_info_[node_name]["size"] >= self.mask:
+                    cover[idxs[pt]].append(node_name)
+
+            clus_base += np.max(clusters) + 1
+
+        # Insert the simplices of the Mapper complex 
+        for i in range(num_pts):
+            self.mapper_.insert(cover[i], filtration=-3)
+        self.mapper_.initialize_filtration()
+
+        return self
+
+
+class CoverComplex(BaseEstimator, TransformerMixin):
+    """
+    Constructor for the CoverComplex class. This class wraps Mapper and Graph Induced complexes in a single interface. Graph Induced complexes can still be called from the class GraphInducedComplex (with a few more functionalities, such as defining datasets or graphs through files). Key differences between Mapper and Graph Induced complexes (GIC) are: Mapper nodes are defined with given input clustering method while GIC nodes are defined with given input graph, GIC accepts partitions instead of covers while Mapper requires cover elements to overlap, Mapper can handle cover defined with multiple filter functions while GIC requires single filter function. 
```

Comment:
CoverComplex documentation explains the difference with GraphInducedComplex. Maybe a comment in GraphInducedComplex should warn the user that he can use CoverComplex would be nice.

Reply:
Yeah you are right I modified the doc